charts/fleet/templates/deployment.yaml

```diff
@@ -1,12 +1,14 @@
-{{ $shards := list "" }}
-{{ if .Values.shards }}
-{{ $shards = concat $shards .Values.shards | uniq }}
-{{ end }}
-{{ range $shards }}
+{{- $shards := list (dict "id" "" "nodeSelector" dict) -}}
+{{- if .Values.shards -}}
+  {{- range .Values.shards -}}
+    {{- $shards = append $shards . -}}
```

Comment:
What if a user specifies the same shard (or at least the same shard ID, possibly with different node selectors) multiple times in `.Values.shards`?

Reply:
An error, I need to work on this, I tried several implementations not working, I'll take time to do it